source/backend/api/Areas/Property/Controllers/SearchController.cs

```diff
@@ -87,6 +88,28 @@ public IActionResult GetProperties([FromBody] PropertyFilterModel filter)
             var result = _mapper.Map<BModel.PageModel<Models.Search.PropertyModel>>(page);
             return new JsonResult(result);
         }
+
+        /// <summary>
+        /// Get all the properties that satisfy the advanced filter parameters.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        [HttpPost("advanced-filter")]
+        [HasPermission(Permissions.PropertyView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(HashSet<long>), 200)]
+        [SwaggerOperation(Tags = new[] { "property" })]
+        public IActionResult FilterProperties([FromBody] PropertyFilterCriteria filter)
+        {
+            filter.ThrowBadRequestIfNull($"The request must include a filter.");
+            if (filter == null)
+            {
+                throw new BadRequestException("Property filter must contain valid values.");
+            }
+
+            var page = _propertyRepository.GetMatchingIds(filter);
```

Comment:
no service for this one?

Reply:
We could rework the property controller to use a service, but right now is contacting the repository directly